iocore/net/SSLNetVConnection.cc

```diff
@@ -508,6 +516,17 @@ SSLNetVConnection::net_read_io(NetHandler *nh, EThread *lthread)
           return;
         }
       }
+      // move over to the socket if we haven't already
+      if (this->handShakeBuffer) {
```

Comment:
if (this->get_context() == NET_VCONNECTION_IN) { maybe better. It means we are only do this on client side.